pkg/plugin/manager.go

```diff
@@ -0,0 +1,157 @@
+package plugin
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/cavaliergopher/grab/v3"
+	"github.com/massalabs/thyra/pkg/config"
+	"github.com/xyproto/unzip"
+)
+
+// Directory return the plugin directory.
+// Note: the plugin directory is the /plugins inside the home directory.
+func Directory() string {
+	homeDir, _ := config.GetConfigDir()
+	pluginsDir := path.Join(homeDir, "my_plugins")
+	_, err := os.Stat(pluginsDir)
+
+	if os.IsNotExist(err) {
+		err := os.MkdirAll(pluginsDir, os.ModePerm)
+		if err != nil {
+			panic(fmt.Errorf("getting plugins directory: creating folder: %w", err))
+		}
+	}
+
+	return pluginsDir
+}
+
+type Manager struct {
+	plugins map[int64]*Plugin
+	m       sync.RWMutex
+	id      int64
+}
+
+func NewManager() *Manager {
+	//nolint:exhaustruct
+	return &Manager{plugins: make(map[int64]*Plugin), id: 0}
+}
+
+// ID returns the list of all the plugin id.
+func (m *Manager) ID() []int64 {
```

Comment:
why i64 ? to avoid conversion on the i ?

Reply:
Because integer in go-swagger are int64 by default.